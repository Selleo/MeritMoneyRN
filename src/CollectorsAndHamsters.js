import React, { Component } from 'react'
import { Text, View, StyleSheet } from 'react-native'

export default class CollectorsAndHamsters extends Component {
  render() {
    return (
      <View style={styles.container}>
        <Text>CollectorsAndHamsters</Text>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  }
})
