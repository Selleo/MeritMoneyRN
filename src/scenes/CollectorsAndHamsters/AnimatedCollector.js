import React, { Component } from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import { Animated } from 'react-native'

export class AnimatedCollector extends Component {
  static propTypes = {
    children: PropTypes.element.isRequired,
    currentTab: PropTypes.string.isRequired,
    style: PropTypes.object.isRequired,
    targetHeight: PropTypes.number.isRequired,
  }

  state = {
    currentHeight: new Animated.Value(0),
  }

  componentDidUpdate = prevProps => {
    const prevTab = prevProps.currentTab
    const { currentTab } = this.props

    if (prevTab && prevTab !== currentTab && currentTab === 'CollectorsAndHamsters') {
      this.animate()
    }

    if (prevTab === 'CollectorsAndHamsters' && currentTab !== 'CollectorsAndHamsters') {
      this.setState({ currentHeight: new Animated.Value(0) })
    }
  }

  animate = () => {
    Animated.timing(this.state.currentHeight, {
      toValue: this.props.targetHeight,
      duration: 1000,
    }).start()
  }

  render() {
    const { currentHeight } = this.state

    return (
      <Animated.View
        style={{
          ...this.props.style,
          height: currentHeight,
        }}
      >
        {this.props.children}
      </Animated.View>
    )
  }
}

const mapStateToProps = ({ currentTab }) => ({ currentTab })

export default connect(mapStateToProps)(AnimatedCollector)
