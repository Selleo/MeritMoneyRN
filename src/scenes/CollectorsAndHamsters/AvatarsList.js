import React, { Component } from 'react'
import { View } from 'react-native'

import Avatar from 'src/components/Avatar'
export default class AvatarsList extends Component {
  render() {
    return (
      <View>
        <Avatar size={100} />
        <Avatar size={100} />
        <Avatar size={100} />
        <Avatar size={100} />
        <Avatar size={100} />
        <Avatar size={100} />
      </View>
    )
  }
}
