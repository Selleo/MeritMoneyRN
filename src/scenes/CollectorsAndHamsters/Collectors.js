import React, { Component } from 'react'
import { StyleSheet, View } from 'react-native'

export default class Collectors extends Component {
  render() {
    return (
      <View style={styles.container} testID="collectorsContainer">
        <View style={styles.firstPlace} />
        <View style={styles.secondPlace} />
        <View style={styles.thirdPlace} />
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 0.5,
    flexDirection: 'row',
    justifyContent: 'space-around',
  },
  thirdPlace: {
    alignSelf: 'flex-end',
    backgroundColor: '#888',
    height: '50%',
    width: 50,
  },
  secondPlace: {
    alignSelf: 'flex-end',
    backgroundColor: '#888',
    height: '75%',
    width: 50,
  },
  firstPlace: {
    alignSelf: 'flex-end',
    backgroundColor: '#888',
    height: '100%',
    width: 50,
  },
})
