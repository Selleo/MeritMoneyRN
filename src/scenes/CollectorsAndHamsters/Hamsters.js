import React, { Component } from 'react'
import { View, Image, StyleSheet } from 'react-native'
import { Avatar } from 'react-native-elements'

export default class Hamsters extends Component {
  render() {
    return (
      <View style={styles.container}>
        <View style={styles.hamstersContainer} testID="hamstersContainer">
          <Avatar
            medium
            rounded
            source={{
              uri: 'https://www.emberjs.com/images/tomsters/teaching-reverse-79f66d70.png',
            }}
          />
        </View>
        <View style={styles.imageContainer}>
          <Image
            source={{
              uri: 'https://www.emberjs.com/images/tomsters/teaching-reverse-79f66d70.png',
            }}
            style={styles.image}
          />
        </View>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 0.5,
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  imageContainer: {
    alignItems: 'flex-end',
  },
  hamstersContainer: {
    flexDirection: 'row',
    flexWrap: 'wrap',
    width: 200,
    margin: 10,
  },
  image: {
    width: 100,
    height: 100,
  },
})
