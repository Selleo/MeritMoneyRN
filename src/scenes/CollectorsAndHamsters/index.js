import React, { Component } from 'react'
import { Text, View, StyleSheet } from 'react-native'
import { Divider } from 'react-native-elements'

import Collectors from './Collectors'
import Hamsters from './Hamsters'

export default class CollectorsAndHamsters extends Component {
  render() {
    return (
      <View style={styles.container}>
        <View style={styles.headerContainer}>
          <Text style={styles.header}>TOP Collectors</Text>
        </View>
        <Collectors />
        <Divider style={styles.divider} />
        <View style={styles.headerContainer}>
          <Text style={styles.header}>Hamsters</Text>
        </View>
        <Hamsters />
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  headerContainer: {
    marginTop: 10,
    justifyContent: 'center',
    alignItems: 'center',
  },
  header: {
    fontSize: 26,
  },
  divider: {
    backgroundColor: 'blue',
    height: 1,
    margin: 10,
  },
})
