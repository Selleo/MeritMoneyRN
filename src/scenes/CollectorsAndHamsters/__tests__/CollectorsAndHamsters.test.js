import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

import CollectorsAndHamsters from '../'

describe('<CollectorsAndHamsters />', () => {
  const wrapper = renderer.create(<CollectorsAndHamsters />)

  test('render', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('_selected', () => {
    test('should return selected state', () => {
      const instance = wrapper.getInstance()
      instance.state.type = 'collectors'
      expect(instance._selected('collectors')).toBe(false)
      expect(instance._selected('hamsters')).toBe(true)
    })
  })

  describe('_selectFilter', () => {
    test('should change selected state', () => {
      const instance = wrapper.getInstance()
      instance.state.type = 'collectors'
      instance._selectFilter('hamsters')()
      expect(instance.state.type).toBe('hamsters')
    })
  })
})
