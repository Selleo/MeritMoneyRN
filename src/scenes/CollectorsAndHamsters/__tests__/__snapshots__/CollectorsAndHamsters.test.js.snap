// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CollectorsAndHamsters /> render 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "paddingTop": 70,
        }
      }
    >
      <View>
        <RNLinearTextGradient
          accessible={true}
          allowFontScaling={true}
          colors={
            Array [
              4282147443,
              4281317757,
            ]
          }
          ellipsizeMode="tail"
          gradientEnd={
            Object {
              "x": 1,
              "y": 0,
            }
          }
          gradientStart={
            Object {
              "x": 0,
              "y": 0,
            }
          }
          locations={
            Array [
              0,
              1,
            ]
          }
          style={
            Array [
              Object {
                "color": "gray",
              },
              Object {
                "fontSize": 30,
              },
            ]
          }
        >
          TOP
        </RNLinearTextGradient>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "paddingHorizontal": 20,
            "paddingVertical": 30,
          }
        }
      >
        <View
          style={
            Object {
              "marginHorizontal": 20,
            }
          }
        >
          View
        </View>
        <View
          style={
            Object {
              "marginHorizontal": 20,
            }
          }
        >
          View
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 3,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "flexWrap": "wrap",
              "justifyContent": "space-between",
              "paddingHorizontal": 20,
            }
          }
        >
          <View
            style={
              Object {
                "paddingHorizontal": 15,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#ffffff",
                  },
                  Object {
                    "borderRadius": 40,
                    "height": 80,
                    "width": 80,
                  },
                ]
              }
            >
              <Image
                source={
                  Object {
                    "uri": "https://i.imgur.com/ckqCXMS.png",
                  }
                }
                style={
                  Array [
                    Object {
                      "borderRadius": 40,
                      "height": 80,
                      "width": 80,
                    },
                    Object {
                      "zIndex": -1,
                    },
                  ]
                }
              />
            </View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#ffffff",
                  "paddingVertical": 5,
                  "textAlign": "center",
                }
              }
            >
              Dominik
              

               Duda
            </Text>
          </View>
          <View
            style={
              Object {
                "paddingHorizontal": 15,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#ffffff",
                  },
                  Object {
                    "borderRadius": 40,
                    "height": 80,
                    "width": 80,
                  },
                ]
              }
            >
              <Image
                source={
                  Object {
                    "uri": "https://i.imgur.com/ckqCXMS.png",
                  }
                }
                style={
                  Array [
                    Object {
                      "borderRadius": 40,
                      "height": 80,
                      "width": 80,
                    },
                    Object {
                      "zIndex": -1,
                    },
                  ]
                }
              />
            </View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#ffffff",
                  "paddingVertical": 5,
                  "textAlign": "center",
                }
              }
            >
              Jak
              

              Smiesz
            </Text>
          </View>
          <View
            style={
              Object {
                "paddingHorizontal": 15,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#ffffff",
                  },
                  Object {
                    "borderRadius": 40,
                    "height": 80,
                    "width": 80,
                  },
                ]
              }
            >
              <Image
                source={
                  Object {
                    "uri": "https://i.imgur.com/ckqCXMS.png",
                  }
                }
                style={
                  Array [
                    Object {
                      "borderRadius": 40,
                      "height": 80,
                      "width": 80,
                    },
                    Object {
                      "zIndex": -1,
                    },
                  ]
                }
              />
            </View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#ffffff",
                  "paddingVertical": 5,
                  "textAlign": "center",
                }
              }
            >
              Bartłomiej
              

              Wójtowicz
            </Text>
          </View>
          <View
            style={
              Object {
                "paddingHorizontal": 15,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#ffffff",
                  },
                  Object {
                    "borderRadius": 40,
                    "height": 80,
                    "width": 80,
                  },
                ]
              }
            >
              <Image
                source={
                  Object {
                    "uri": "https://i.imgur.com/ckqCXMS.png",
                  }
                }
                style={
                  Array [
                    Object {
                      "borderRadius": 40,
                      "height": 80,
                      "width": 80,
                    },
                    Object {
                      "zIndex": -1,
                    },
                  ]
                }
              />
            </View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#ffffff",
                  "paddingVertical": 5,
                  "textAlign": "center",
                }
              }
            >
              Grzegorz
              

              aaaaa
            </Text>
          </View>
          <View
            style={
              Object {
                "paddingHorizontal": 15,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#ffffff",
                  },
                  Object {
                    "borderRadius": 40,
                    "height": 80,
                    "width": 80,
                  },
                ]
              }
            >
              <Image
                source={
                  Object {
                    "uri": "https://i.imgur.com/ckqCXMS.png",
                  }
                }
                style={
                  Array [
                    Object {
                      "borderRadius": 40,
                      "height": 80,
                      "width": 80,
                    },
                    Object {
                      "zIndex": -1,
                    },
                  ]
                }
              />
            </View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#ffffff",
                  "paddingVertical": 5,
                  "textAlign": "center",
                }
              }
            >
              Panie
              

              Janie
            </Text>
          </View>
          <View
            style={
              Object {
                "paddingHorizontal": 15,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#ffffff",
                  },
                  Object {
                    "borderRadius": 40,
                    "height": 80,
                    "width": 80,
                  },
                ]
              }
            >
              <Image
                source={
                  Object {
                    "uri": "https://i.imgur.com/ckqCXMS.png",
                  }
                }
                style={
                  Array [
                    Object {
                      "borderRadius": 40,
                      "height": 80,
                      "width": 80,
                    },
                    Object {
                      "zIndex": -1,
                    },
                  ]
                }
              />
            </View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#ffffff",
                  "paddingVertical": 5,
                  "textAlign": "center",
                }
              }
            >
              Rano
              

              Wstań
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
